BackEnd/src/main/kotlin/cub/trackmyoffer/AuthRouting.kt

```diff
@@ -0,0 +1,94 @@
+package cub.trackmyoffer
+
+
+import io.ktor.client.statement.*
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.netty.*
+import io.ktor.server.plugins.cors.routing.*
+import io.ktor.server.routing.*
+import io.ktor.client.call.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.*
+import io.ktor.server.auth.*
+import io.ktor.server.html.*
+import io.ktor.server.response.*
+import io.ktor.server.sessions.*
+import io.ktor.serialization.kotlinx.json.*
+import io.ktor.server.request.*
+import kotlinx.html.*
+import kotlinx.serialization.*
+
+fun Route.authRouting(httpClient: HttpClient) {
+
+    authenticate("google-oauth") {
+        get("/login") {
+            // Redirect to Google OAuth
+        }
+
+        get("/callback") {
+            val currentPrincipal: OAuthAccessTokenResponse.OAuth2? = call.authentication.principal()
+
+            currentPrincipal?.let { principal ->
+                principal.state?.let { state ->
+                    call.sessions.set(UserSession(state, principal.accessToken))
+                    call.respondRedirect("/home")
+                    return@get // Ensure no further response is sent
+                }
+            }
+
+            call.respondRedirect("/login")
+        }
+    }
+
+    get("/home") {
+        val userSession: UserSession? = getSession(call)
+        if (userSession != null) {
+            val userInfo: UserInfo = getPersonalGreeting(httpClient, userSession)
+//            call.respondText("Hello, ${userInfo.name}! Welcome home!")
+            call.respondRedirect("http://localhost:5000/")
```

Comment:
Please put front host & port into application.conf as it is currently with feature provider, even if this is temporal